### PR TITLE
Implemented `chunks` to retrieve references in batch

### PR DESCRIPTION
**Pull request by JelleBoersma**
Fixes #<issue>

## Checklist:
- [x] My code follows the pykechain style
    - [ ] I added type hinting to all functions
    - [ ] I added documentation for all public functions
    - [ ] I locally used `tox` to check for dists and docs errors
    - [ ] My code passes the `pep8` and `flake8` linting checks and no warnings
    - [ ] I removed unused imports, using `Code > Optimize Imports` in PyCharm
    - [ ] I used [`black`](https://github.com/psf/black) to format the new code
- [x] I have added tests that prove my fix is effective or that my feature works
    - [ ] I committed fresh test cassettes
    - [ ] I have proper test coverage (don't decline the coverage of the test)
- [x] I asked another teammate to review the code
- [x] I update the Changelog.md with the appropriate changes.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,6 +1,8 @@
 Change Log
 ==========
 
+* :bug: Retrieving the `value` of any reference property is now performed in batches to limit request size, using the new `chunks()` method.
```
Comment (jberends) [inline CHANGELOG.rst:4]:
```suggestion
* :+1: Retrieving the `value` of any reference property is now performed in batches to limit request size, using the new `chunks()` method.
```

File: pykechain/defaults.py [hunk 2]
```diff
@@ -31,6 +31,10 @@
 # If the backoff_factor is 0.1, then sleep() will sleep for [0.0s, 0.2s, 0.4s, …] between retries.
 RETRY_BACKOFF_FACTOR = 0.8
 
+
+# Batching of parts when a large number of parts are requested at once
+PARTS_BATCH_LIMIT = 100  # number of parts
```
Comment (jberends) [inline pykechain/defaults.py:36]: Could we 'reuse' the batch retrieval numbers of the part retrieval in batches? Less 'thresholds', more 'consistent'.

File: pykechain/models/base_reference.py [hunk 3]
```diff
@@ -90,6 +91,21 @@ def _retrieve_objects(self, **kwargs) -> List[Base]:
         """
         pass  # pragma: no cover
 
+    @staticmethod
+    def chunks(lst: List[Any], n: Optional[int] = PARTS_BATCH_LIMIT) -> List:
+        """
+        Yield successive n-sized chunks from lst.
```
Comment (jberends) [inline pykechain/models/base_reference.py:97]:
```suggestion
        Yield successive n-sized chunks from a list.
```

File: pykechain/models/property_multi_reference.py [hunk 4]
```diff
@@ -39,11 +39,13 @@ def _retrieve_objects(self, **kwargs) -> List[Part]:
...
                     models = self.model().value
                 if models:
-                    parts = list(self._client.parts(
-                        id__in=','.join(part_ids),
-                        model=models[0],
-                        category=None,
-                    ))
+                    parts = list()
+                    for chunk in self.chunks(part_ids):
+                        parts += list(self._client.parts(
```
Comment (jberends) [inline pykechain/models/property_multi_reference.py:44]: `parts.extend()` is faster than concatenation. 

```suggestion
                        parts.extend(
                            list(self._client.parts(
```

File: pykechain/models/property_reference.py [hunk 5]
```diff
@@ -49,7 +49,9 @@ def _retrieve_objects(self, **kwargs) -> List[Scope]:
 
         scopes = []
         if scope_ids:
-            scopes = list(self._client.scopes(id__in=",".join(scope_ids), status=None))
+            scopes = list()
+            for chunk in self.chunks(scope_ids):
+                scopes += list(self._client.scopes(id__in=",".join(chunk), status=None))
```
Comment (jberends) [inline pykechain/models/property_reference.py:54]: scopes.extend() is faster than concatenation.

```suggestion
                scopes.extend(list(self._client.scopes(id__in=",".join(chunk), status=None)))
```

File: pykechain/models/property_reference.py [hunk 6]
```diff
@@ -144,7 +146,9 @@ def _retrieve_objects(self, **kwargs) -> List[user.User]:
 
         users = []
         if user_ids:
-            users = list(self._client.users(id__in=",".join(user_ids)))
+            users = list()
+            for chunk in self.chunks(user_ids):
+                users = list(self._client.users(id__in=",".join(chunk)))
```
Comment (jberends) [inline pykechain/models/property_reference.py:151]: does not concatenate the list (possible bug?)

`extend()` is faster than concatenation

```suggestion
                users.extend(list(self._client.users(id__in=",".join(chunk))))
```

**Review by jberends — CHANGES_REQUESTED** [review]

File: pykechain/defaults.py (hunk 2 above)
Comment (JelleBoersma) [reply pykechain/defaults.py:36]: Already is?

**Review by jberends — APPROVED** [review]

**Pull request merged**